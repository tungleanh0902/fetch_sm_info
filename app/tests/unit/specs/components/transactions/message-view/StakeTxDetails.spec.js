import { shallowMount } from "@vue/test-utils"
import StakeTxDetails from "src/components/transactions/message-view/StakeTxDetails"

describe(`StakeTxDetails`, () => {
  let wrapper, $store

  $store = {
    getters: {
      network: `cosmos-hub-testnet`,
    },
  }

  const tx = {
    type: "StakeTx",
    hash: "3CA728671B8078E71697B62237AD694052779F80B56880F6A6F1702F53EA3081",
    height: 308453,
    timestamp: "2020-01-10T09:02:54Z",
    memo: "",
    success: true,
    fees: [
      {
        denom: "ATOM",
        amount: "0.00045",
      },
    ],
    details: {
      to: ["cosmosvaloper18ymm350peujvq2xy9ymyqj4v34ekvnk3tsekuz"],
      amount: {
        denom: "ATOM",
        amount: "10",
      },
    },
  }

  it(`renders a stake transaction message`, () => {
    wrapper = shallowMount(StakeTxDetails, {
      propsData: {
        transaction: tx,
        validators: {},
      },
      mocks: {
        $store,
      },
      stubs: [`router-link`],
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
