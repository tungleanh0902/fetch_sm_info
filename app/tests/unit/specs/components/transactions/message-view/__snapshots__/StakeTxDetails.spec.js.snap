// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StakeTxDetails renders a stake transaction message 1`] = `
<div
  class="tx__content"
>
  <transactionicon-stub
    transactiontype="Staked"
  />
   
  <div
    class="tx__content__left"
  >
    <h3>
      Staked
    </h3>
     
    <span>
      With 
    </span>
     
    <router-link-stub
      to="[object Object]"
    >
      <!---->
      
      cosmosvaloper…ekuz
    
    </router-link-stub>
  </div>
   
  <div
    class="tx__content__right"
  >
    <p
      class="amount"
    >
      
      10 
      ATOM
    
    </p>
  </div>
</div>
`;
